Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM golang:1.12 as builder
+
+WORKDIR /go/src/github.com/cloudflare/cloudflared/
+COPY . .
+RUN apt-get update && apt-get install -y --no-install-recommends upx
+RUN make cloudflared
+RUN upx --no-progress cloudflared
+
+FROM gcr.io/distroless/base
+COPY --from=builder /go/src/github.com/cloudflare/cloudflared/cloudflared /usr/local/bin/
+ENTRYPOINT ["cloudflared"]
```

Comment:
cloudflared has an auto-update feature; it can be disabled by running with the `--no-autoupdate` flag. Do you think that's worth adding to the entrypoint? I think that might make sense: as a user of a container, I expect it not to rebuild itself at runtime...

Reply:
Yeah, I think this would make sense, especially considering we're versioning containers based off of versions/tags. We would want it to stay whatever version we explicitly create.

Reply:
> @nickvollmar said: cloudflared has an auto-update feature; it can be disabled by running with the --no-autoupdate flag.

I was leaning on this output from the cloudflared log: 
```console
INFO[0000] cloudflared will not automatically update when run from the shell. To enable auto-updates, run cloudflared as a service: https://developers.cloudflare.com/argo-tunnel/reference/service/ 
```

..but there's definitely a point in being explicit here, especially if behavior changes in `cloudflared`.